aiida_testing/export_cache/_fixtures.py

```diff
@@ -0,0 +1,609 @@
+
+        :param path_to_cache: str or path to the AiiDA export file to load,
+                              if path_to_cache points to a directory,
+                              all import files in this dir are imported
+        :param node: AiiDA node, determines which cache to load (CURRENTLY NOT IMPLEMENTED)
+        :raises : OSError or FileNotFoundError, if import file is non existent
+        """
+        if path_to_cache is None:  #pylint: disable=no-else-raise
+            if node is None:
+                raise ValueError(
+                    "Node argument can not be None "
+                    "if no explicit 'path_to_cache' is specified"
+                )
+            raise NotImplementedError('The node argument is currently not supported')
+            #else:  # create path from node
+            #    pass
+            #    # get default data dir
+            #    # get hash for give node
+            #    # construct path from that
+        else:
+            # relative paths given will be completed with cwd
+            full_import_path = absolute_archive_path(path_to_cache)
+
+        if os.path.exists(full_import_path):
+            if os.path.isfile(full_import_path):
+                # import cache, also import extras
+                import_with_migrate(full_import_path)
+            elif os.path.isdir(full_import_path):
+                for filename in os.listdir(full_import_path):
+                    file_full_import_path = os.path.join(full_import_path, filename)
+                    # we curretly assume all files are valid aiida exports...
+                    # maybe check if valid aiida export, or catch exception
+                    import_with_migrate(file_full_import_path)
+            else:  # Should never get there
+                raise OSError(
+                    f"Path: {full_import_path} to be imported exists, but is neither a file or directory."
+                )
+        else:
+            raise FileNotFoundError(f"File: {full_import_path} to be imported does not exist.")
+
+        # need to rehash after import, otherwise cashing does not work
+        # for this we rehash all process nodes
+        # this way we use the full caching mechanism of aiida-core.
+        # currently this should only cache CalcJobNodes
+        _rehash_processes()
+
+    return _load_cache
+
+
+@pytest.fixture(scope='function')
+def with_export_cache(export_cache, load_cache, absolute_archive_path):
+    """
+    Fixture to use in a with block
+    - Before the block the given cache is loaded (if it exists)
+    - within this block the caching of AiiDA is enabled.
+    - At the end an AiiDA export can be created (if test data should be overwritten)
+    Requires an absolute path to the export file to load or export to.
+    Export the provenance of all calcjobs nodes within the test.
+    """
+
+    @contextmanager
+    def _with_export_cache(data_dir_abspath, calculation_class=None, overwrite=False):
```

Comment:
Why limit this only to a single calculation class?

I guess in practice you're anyhow going to target all calculation classes?
Would that make sense as a default?